src/GenFacades/Program.cs

```diff
@@ -459,23 +488,31 @@ private INamespaceTypeDefinition GetSeedType(string docId, IReadOnlyList<INamesp
                 string preferredSeedAssembly;
                 if (_seedTypePreferences.TryGetValue(docId, out preferredSeedAssembly))
                 {
-                    return seedTypes.SingleOrDefault(t => String.Equals(t.ContainingUnitNamespace.Unit.Name.Value, preferredSeedAssembly, StringComparison.OrdinalIgnoreCase));
+                    return seedTypes.SingleOrDefault(t => String.Equals(GetContainingUnitNamespaceFromType(t), preferredSeedAssembly, StringComparison.OrdinalIgnoreCase));
                 }
 
                 return null;
             }
 
-            private static void TraceDuplicateSeedTypeError(string docId, IReadOnlyList<INamespaceTypeDefinition> seedTypes)
+            private static void TraceDuplicateSeedTypeError(string docId, IReadOnlyList<INamedTypeDefinition> seedTypes)
             {
                 Trace.TraceError("The type '{0}' is defined in multiple seed assemblies. If this is intentional, specify one of the following arguments to choose the preferred seed type:", docId);
 
-                foreach (INamespaceTypeDefinition type in seedTypes)
+                foreach (INamedTypeDefinition type in seedTypes)
                 {
-                    Trace.TraceError("  /preferSeedType:{0}={1}", docId.Substring("T:".Length), type.ContainingUnitNamespace.Unit.Name.Value);
+                    Trace.TraceError("  /preferSeedType:{0}={1}", docId.Substring("T:".Length), GetContainingUnitNamespaceFromType(type));
                 }
             }
 
-            private void AddTypeForward(Assembly assembly, INamedTypeDefinition seedType)
+            private static string GetContainingUnitNamespaceFromType(INamedTypeDefinition type)
+            {
+                INamespaceTypeDefinition namespaceTypeDefinition = type as INamespaceTypeDefinition;
+                if (namespaceTypeDefinition != null)
+                    return namespaceTypeDefinition.ContainingUnitNamespace.Unit.Name.Value;
+                return type.GetAssembly().Name.Value;
+            }
+
+            private void AddTypeForward(Assembly assembly, INamedTypeDefinition seedType, IAssembly contractAssembly)
             {
                 var alias = new NamespaceAliasForType();
```

Comment:
Does this need to be something other than NamespaceAliasForType in the nested case?